src/App.tsx

```diff
@@ -58,203 +57,100 @@ const PageNumber = styled.div`
   color: white;
 `;
 
-export type MovieResponse = {
-  Search: Movie[];
-  Response: string;
-  totalResults: string;
-};
-
-export type Movie = {
-  Title: string;
-  Year: string;
-  imdbID: string;
-  Type: string;
-  Poster: string;
-  imdbRating: string;
-};
-
-export type DetailedMovie = {
-  Title: string;
-  Plot: string;
-  Released: string;
-  Runtime: string;
-  Awards: string;
-  imdbID: string;
-  imdbRating: string;
-  Poster: string;
-};
-
-export type AppState = {
-  search: string;
-  result: MovieResponse | undefined;
-  page: number;
-  selectedMovie: DetailedMovie | undefined;
-};
-
-const initialDetailedState: DetailedMovie = {
-  Title: "",
-  Plot: "",
-  Released: "",
-  Runtime: "",
-  Awards: "",
-  imdbID: "",
-  imdbRating: "",
-  Poster: "",
-};
-
-const initialState: AppState = {
-  search: "Star Wars",
-  result: undefined,
-  page: 1,
-  selectedMovie: undefined,
-};
-
-export type AppEvents =
-  | {
-      type: "search set";
-      payload: { search: string };
-    }
-  | {
-      type: "movie response set";
-      movieResPayload: MovieResponse;
-    }
-  | {
-      type: "page number changed";
-      pagePayload: { page: number };
-    }
-  | {
-      type: "select movie response set";
-      selMovieResPayload: DetailedMovie | undefined;
-    };
-
-export const reducer: React.Reducer<AppState, AppEvents> = (state, event) => {
-  switch (event?.type) {
-    case "search set": {
-      return {
-        ...state,
-        search: event.payload.search,
-      };
-    }
-    case "movie response set": {
-      return { ...state, result: event.movieResPayload };
-    }
-    case "page number changed": {
-      return {
-        ...state,
-        page: event.pagePayload.page,
-      };
-    }
-    case "select movie response set": {
-      return {
-        ...state,
-        selectedMovie: event.selMovieResPayload,
-      };
-    }
-  }
-};
-
 function App() {
-  const [state, update] = useReducer(reducer, initialState);
-  const [selectedID, updateSelectedID] = React.useState("");
-  const [popUpState, updatePopUpState] = React.useState(false);
-  const [mediaType, setMediaType] = React.useState("movie");
+  const [state, update] = useReducer(reducer, initialState());
 
   useEffect(() => {
-    async function callToAPIs() {
+    async function fetchMovies() {
       try {
-        const [movieResponse] = await Promise.all([
-          getMovies(state.search, state.page, mediaType),
-          getSelected(selectedID),
-        ]);
-        update({ type: "movie response set", movieResPayload: movieResponse });
+        // destructure and alias so Search can be referred to as movies
```

Comment:
Still don't quite understand this, if you didn't destructure would you not be able to access .search .page .mediaType explicitly?

Reply:
Destructuring is equal to standard access

E.g.

```typescript

const test = demo.person.name

// Equal to

const { person: { name: test } } = demo

But you don't need to alias `{ name: test}` you could just leave it as `{ name }` and access as `name`.

```

Reply:
Ah gotcha!